src/Controllers/ApiController.php

```diff
@@ -40,40 +38,16 @@ protected function callApi(): mixed
             return $this->callApiCommon($madelineProto);
         }
 
-        //GROUP REQUESTS IN BULKS
-        /** @var ?DeferredFuture $lock */
-        static $lock = null;
-
-        if (!$lock) {
-            try {
-                $lock = new DeferredFuture();
-                delay($this->waitNextTick());
-                $lock->complete();
-            } finally {
-                $lock = null;
-            }
-        } else {
-            $lock->getFuture()->await();
+        if (!self::$w) {
+            $f = new DeferredFuture;
+            self::$w = $f->getFuture();
+            EventLoop::delay(0.001, static function () use ($f): void {
```

Comment:
Lets use delay from config instead of constant.

Reply:
Very bad idea imo

Reply:
By default its 0.0 in config.